### PR TITLE
Added webhook ping message support

### PR DESCRIPTION
**Pull request by hortha**
### Context

Today, in order to end-to-end test webhook delivery processing we have to change a user in order to generate a UserUpdated message. In production, this is undesirable and even in non-production it’s a little messy. We should add a ‘Ping’ message that can be triggered from ID’s admin UI.

### Changes proposed in this pull request

Add a PingMessage to Notifications.Messages with a WebHookId property (that is excluded from JSON serialization).

Modify the GetWebHooksForNotification method so that only the WebHook matching the PingMessage’s WebHookId is returned.

Note that enabling/disabling this message type should not be configurable via the UI.

Add a Ping button to the Admin/EditWebHook page. When clicked, this should generate a NotificationEnvelope containing a PingMessage for the current WebHookId then dispatch it (using INotificationPublisher.PublishNotification).

### Guidance to review

Relatively simple change.

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [x] Cleaned commit history
-   [ ] Tested by running locally
-   [ ] Reminder created to manually clean any removed app settings post deployment

### CODE REVIEW

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/EditWebHook.cshtml [hunk 1]
```diff
@@ -23,6 +23,8 @@
         <govuk-checkboxes-item value="@true">Enabled</govuk-checkboxes-item>
     </govuk-checkboxes>
 
+    <govuk-button type="submit" class="govuk-button--secondary" asp-page-handler="Ping" value="ping">Ping</govuk-button>
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/EditWebHook.cshtml:26]: Do we need the `value` attribute here?

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/EditWebHook.cshtml [hunk 2]
```diff
@@ -39,5 +41,5 @@
         <govuk-checkboxes-item value="@true">Regenerate secret</govuk-checkboxes-item>
     </govuk-checkboxes>
 
-    <govuk-button type="submit">Save</govuk-button>
+    <govuk-button type="submit" asp-page-handler="Save" value="save">Save</govuk-button>
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/EditWebHook.cshtml:44]: Can we keep this as the regular page handler method i.e. no `asp-page-handler`?

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/EditWebHook.cshtml.cs [hunk 3]
```diff
@@ -123,7 +132,34 @@ public async Task<IActionResult> OnPost()
...
+        };
+
+        try
+        {
+            await _notificationPublisher.PublishNotification(pingNotification);
+        }
+        catch (Exception exception)
+        {
+            ModelState.AddModelError(nameof(Endpoint), $"Failed sending message: {exception.Message}");
+            var webHook = await _dbContext.WebHooks.SingleOrDefaultAsync(wh => wh.WebHookId == WebHookId);
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/EditWebHook.cshtml.cs:160]:
```suggestion
            var webHook = await _dbContext.WebHooks.SingleAsync(wh => wh.WebHookId == WebHookId);
```

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/EditWebHook.cshtml (hunk 1 above)
Comment (hortha) [reply dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/EditWebHook.cshtml]: removed

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/EditWebHook.cshtml.cs (hunk 3 above)
Comment (hortha) [reply dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/EditWebHook.cshtml.cs]: Done

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/EditWebHook.cshtml (hunk 2 above)
Comment (hortha) [reply dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/EditWebHook.cshtml]: Done

**Review by gunndabad — APPROVED** [review]

**Pull request merged**